.hooks/pre-commit

```diff
@@ -1,4 +1,4 @@
-#/bin/sh
+#!/bin/bash
```

Comment:
Can you make this bin/sh?

Reply:
The script does not work if I am calling it with sh

Reply:
What OS are you on?

Reply:
I am using PopOs (which is basically Ubuntu) with fish shells.

Reply:
Okay, so it doesn't work because you're not on a POSIX compliant shell. In general, we want to keep using sh because it's lighter-weight and maximally portable.

Can you check what other folks have done when using fish with git hooks?

Reply:
The `#!` will inform what shell the script should run in, i.e. sh/bash. So using fish should not be a problem. 

If I understand correctly, running the hook is executing the script. When I am running `sh .git/hooks/pre-commit`, it gives me the following error: 
```
.git/hooks/pre-commit: 3: function: not found
.git/hooks/pre-commit: 5: shift: can't shift that many
```

Reply:
I did some look-up. On Ubuntu, by default, it points your sh to dash, so there is a chance that you might be running dash, but you don't know about it.

Reply:
Yeah, that is the default on Ubuntu although not on MacOS. What does your `sh` location link to (`command -v sh`)?

Reply:
It links to my `/usr/bin/sh`, and `readlink -f $(which sh)` links to `/usr/bin/dash`.

MacOS points `sh` to `bash` (ref)[https://scriptingosx.com/2020/06/about-bash-zsh-sh-and-dash-in-macos-catalina-and-beyond/]. And I have validated it on my Mac.